File: app/src/main/java/com/sp/base/util/SharedPref.java
package com.sp.base.util;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

/**
 * SharedPref class is used for saving data into sharedPreference
 */
public class SharedPref {
    private static SharedPref mInstance = null;
    private static SharedPreferences mSharedPref;
    private Context mContext;

    // use getApplicationContext() to call this method
    public static SharedPref getInstance(Context context) {
        if (mInstance == null) {
            mInstance = new SharedPref(context);
        }
        return mInstance;
    }

    private SharedPref(Context context) {
        mContext = context;
        mSharedPref = context.getSharedPreferences(context.getPackageName(), Activity.MODE_PRIVATE);
    }

    public String read(String key, String defValue) {
        return mSharedPref.getString(key, defValue);
    }

    public int read(String key, int defValue) {
        return mSharedPref.getInt(key, defValue);
    }

    public boolean read(String key, boolean defValue) {
        return mSharedPref.getBoolean(key, defValue);
    }

    public void write(String key, String value) {
        try {
            SharedPreferences.Editor prefsEditor = mSharedPref.edit();
            prefsEditor.putString(key, value);
            prefsEditor.apply();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void write(String key, int value) {
        try {
            SharedPreferences.Editor prefsEditor = mSharedPref.edit();
            prefsEditor.putInt(key, value);
            prefsEditor.apply();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    public void write(String key, boolean value) {
        try {
            SharedPreferences.Editor prefsEditor = mSharedPref.edit();
            prefsEditor.putBoolean(key, value);
            prefsEditor.apply();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: app/src/main/java/com/baseapplication/util/SharedPref.kt
package com.baseapplication.util

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences

/**
 * SharedPref class is used for saving data into sharedPreference
 */
class SharedPref private constructor(private val mContext: Context) {
    init {
        mSharedPref =
            mContext.getSharedPreferences(mContext.getPackageName(), Activity.MODE_PRIVATE)
    }

    fun read(key: String?, defValue: String?): String? {
        return mSharedPref.getString(key, defValue)
    }

    fun read(key: String?, defValue: Int): Int {
        return mSharedPref.getInt(key, defValue)
    }

    fun read(key: String?, defValue: Boolean): Boolean {
        return mSharedPref.getBoolean(key, defValue)
    }

    fun write(key: String?, value: String?) {
        try {
            val prefsEditor: SharedPreferences.Editor = mSharedPref.edit()
            prefsEditor.putString(key, value)
            prefsEditor.apply()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun write(key: String?, value: Int) {
        try {
            val prefsEditor: SharedPreferences.Editor = mSharedPref.edit()
            prefsEditor.putInt(key, value)
            prefsEditor.apply()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    fun write(key: String?, value: Boolean) {
        try {
            val prefsEditor: SharedPreferences.Editor = mSharedPref.edit()
            prefsEditor.putBoolean(key, value)
            prefsEditor.apply()
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    companion object {
        private var mInstance: SharedPref? = null
        private lateinit var mSharedPref: SharedPreferences

        // use getApplicationContext() to call this method
        fun getInstance(context: Context): SharedPref? {
            if (mInstance == null) {
                mInstance = SharedPref(context)
            }
            return mInstance
        }
    }
}
